Packs/Base/Scripts/CommonServerPowerShell/CommonServerPowerShell.Tests.ps1

```diff
@@ -179,3 +179,36 @@ Describe 'Check-UtilityFunctions' {
         }
     }
 }
+Describe "Test ParseDateRange"{
+        It "Naive Tests" -Tag naive{
+            $m_date = Get-Date "2020-11-30T23:30:00.0000000"
+            $three_mins = $m_date.AddMinutes(-3)
+            $three_hours = $m_date.AddHours(-3)
+            $three_days = $m_date.AddDays(-3)
+            $three_weeks = $m_date.AddDays(-7 * 3)
+            $three_months = $m_date.AddMonths(-3)
+            $three_years = $m_date.AddYears(-3)
+            Mock -CommandName Get-Date -MockWith {$m_date}
+            ParseDateRange("3 minutes") | Should -Be @($three_mins, $m_date)
+            ParseDateRange("3 minute") | Should -Be @($three_mins, $m_date)
+            ParseDateRange("3 hours") | Should -Be @($three_hours, $m_date)
+            ParseDateRange("3 hour") | Should -Be @($three_hours, $m_date)
+            ParseDateRange("3 days") | Should -Be @($three_days, $m_date)
+            ParseDateRange("3 day") | Should -Be @($three_days, $m_date)
+            ParseDateRange("3 week") | Should -Be @($three_weeks, $m_date)
+            ParseDateRange("3 weeks") | Should -Be @($three_weeks, $m_date)
+            ParseDateRange("3 month") | Should -Be @($three_months, $m_date)
+            ParseDateRange("3 months") | Should -Be @($three_months, $m_date)
+            ParseDateRange("3 year") | Should -Be @($three_years, $m_date)
+            ParseDateRange("3 years") | Should -Be @($three_years, $m_date)
```

Comment:
Add a case with the bad inputs that will check the exception's string to see that you have fixed it.